assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java

```diff
@@ -12812,9 +12862,19 @@ public Time getCloseTime()
 		 * @return
 		 */
 		public String getAnonymousSubmissionId() {
-				String anonTitle = rb.getString("grading.anonymous.title") ;
-				return this.getId().substring(27) + " (" + anonTitle + ")";
-		}
+			String id = this.getId().substring(27);
+			try {
+				Site site = SiteService.getSite( this.getAssignment().getContext() );
+				if(candidateDetailProvider.useInstitutionalAnonymousId(site)) {
+					id = candidateDetailProvider.getCandidateID(UserDirectoryService.getUser(this.getSubmitterId()), site).orElse(this.getId().substring(27));
+				}
+			} catch(IdUnusedException e){
+				M_log.debug("Site not found " + this.getAssignment().getContext() + " : " + e.getMessage());
```

Comment:
This isn't expected is it? Something has gone pretty wrong if the site can't be found, so shouldn't we log this at a warning level with a stack trace?